### PR TITLE
List elected in "espaces référent" and LRE using new geo zone system

### PR DESCRIPTION
**Pull request by renanbr**
Depends on:

- [x] #4966

### CODE REVIEW

File: src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php [hunk 1]
```diff
@@ -240,46 +247,26 @@ private function withActiveMandatesCondition(QueryBuilder $qb, string $alias = '
...
-            $zoneCondition->add('tag.code IN (:districtDptCodes)');
-            $qb->setParameter('districtDptCodes', $districtDptCodes);
+        if ($zones) {
+            $qb
+                ->innerJoin('mandate.geoZone', 'geo_zone')
+                ->leftJoin('geo_zone.children', 'geo_zone_child')
+                ->andWhere(
+                    $qb->expr()->orX(
+                        $qb->expr()->in('geo_zone.id', ':zones'),
+                        $qb->expr()->in('geo_zone_child.id', ':zones'),
```
Comment (illusionOfParadise) [inline src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php]: We need to check if parent zones are in the managed zoned and not children.
For exemple, if managed zone is department we should find all mandates in city, EPCI and department zone.

File: src/Repository/Geo/ZoneRepository.php [hunk 2]
```diff
@@ -51,4 +52,24 @@ public function zoneableAsZone(ZoneableInterface $zoneable): Zone
 
         return $zone;
     }
+
+    /**
+     * @param Zone[] $zones
```
Comment (illusionOfParadise) [inline src/Repository/Geo/ZoneRepository.php:57]: Wrong return type.

File: src/Repository/Geo/ZoneRepository.php [hunk 3]
```diff
@@ -51,4 +52,24 @@ public function zoneableAsZone(ZoneableInterface $zoneable): Zone
...
+    public function createSelectByParentsQueryBuilder(array $zones): QueryBuilder
+    {
+        $qb = $this->createQueryBuilder('zone');
+
+        return $qb
+            ->leftJoin('zone.children', 'children')
+            ->andWhere(
+                $qb->expr()->orX(
+                    $qb->expr()->in('zone.id', ':zones'),
+                    $qb->expr()->in('children.id', ':zones'),
```
Comment (illusionOfParadise) [inline src/Repository/Geo/ZoneRepository.php:68]: We need to check parents in the zone, not children.
For exemple, if a department is passed to the function, we need to get all boroughs, cities, canton etc and department itself.

**Review by illusionOfParadise — CHANGES_REQUESTED** [review]

File: src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php (hunk 1 above)
Comment (renanbr) [reply src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php]: thanks, fixed in 15eba7ac37d2636b56e493269e0ec3ae10fcf54a

File: src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php [hunk 4]
```diff
@@ -67,14 +71,17 @@ public function countForReferentTags(array $referentTags): int
         ;
     }
 
-    public function isInReferentManagedArea(ElectedRepresentative $electedRepresentative, array $referentTags): bool
+    /**
+     * @param Zone[] $zones
```
[the remark this answers is not in the document]
Comment (renanbr) [reply src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php:75]: done via 8a4661ad815ab0e40f0a51d2a1f333acd8db10d4

File: src/Repository/Geo/ZoneRepository.php (hunk 2 above)
Comment (renanbr) [reply src/Repository/Geo/ZoneRepository.php]: done via 239bb7948be8369f930b3e7a0c58b9252b4ada1d

File: src/Repository/Geo/ZoneRepository.php (hunk 3 above)
Comment (renanbr) [reply src/Repository/Geo/ZoneRepository.php]: done in 239bb7948be8369f930b3e7a0c58b9252b4ada1d

File: src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (renanbr) [reply src/Repository/ElectedRepresentative/ElectedRepresentativeRepository.php]: done f1626025e7edf87155536c980bb0deb688568f74

File: src/Controller/EnMarche/ElectedRepresentative/DeputyElectedRepresentativeController.php [hunk 5]
```diff
@@ -17,8 +17,18 @@ protected function getSpaceType(): string
...
+        if (!$managedDistrict) {
+            return [];
+        }
+
+        $referentTag = $managedDistrict->getReferentTag();
+        if (!$referentTag) {
+            return [];
+        }
+
+        return [$referentTag->getZone()];
```
Comment (ottaviano) [inline src/Controller/EnMarche/ElectedRepresentative/DeputyElectedRepresentativeController.php:32]: why do we have all these checks? 
we can just return `return [$adherent->getManagedDistrict()->getReferentTag()->getZone()];`, no?

File: src/Repository/Geo/ZoneRepository.php [hunk 6]
```diff
@@ -62,4 +62,21 @@ public function createSelectForCandidatesQueryBuilder(): QueryBuilder
             ])
         ;
     }
+
+    public function createSelectByParentsQueryBuilder(array $zones): QueryBuilder
```
Comment (ottaviano) [inline src/Repository/Geo/ZoneRepository.php:66]: seems to be unused

**Review by ottaviano — APPROVED** [review]

**Pull request merged**